lib/puppet/util/uri.rb

```diff
@@ -0,0 +1,18 @@
+# This module holds util funtions for URI's
+module Puppet::Util::Uri
```

Comment:
Can you move this to `Puppet::Network::Uri`? One weird thing about puppet is that it includes [`Puppet::Util`](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider.rb#L38) into every provider, so this module is also added to each provider class. We recently had some crazy conflicts with `Puppet::Util::Package`.

```
irb(main):013:0> pp Puppet::Type::User::ProviderDirectoryservice.constants.sort
[:AbsolutePathPosix,
 :AbsolutePathWindows,
 :AixObject,
 :Autoload,
 :CharacterEncoding,
 :Checksums,
 :ClassGen,
 :Colors,
 :Command,
 :CommandDefiner,
 :CommandLine,
 :Connection,
 :ConstantInflector,
 :DEFAULT_POSIX_MODE,
 :DEFAULT_WINDOWS_MODE,
 :Docs,
 :Errors,
 :Execution,
 :ExecutionStub,
 :Feature,
 :FileNamespaceAdapter,
 :FileWatcher,
 :HttpProxy,
 :InstanceLoader,
 :Json,
 :Ldap,
 :Libuser,
 :Limits,
 :Lockfile,
 :Log,
 :Logging,
 :Metric,
 :ModuleDirectoriesAdapter,
 :MonkeyPatches,
 :MultiMatch,
 :NameService,
 :POSIX,
 :Package,
 :Pidlock,
 :Platform,
 :Plist,
 :Profiler,
 :ProviderFeatures,
 :PsychSupport,
 :RFC_3986_URI_REGEX,
 :RubyGems,
 :RunMode,
 :SSL,
 :SUIDManager,
 :SkipTags,
 :Storage,
 :SymbolicFileMode,
 :TagSet,
 :Tagging,
 :UnixRunMode,
 :Warnings,
 :WatchedFile,
 :Watcher,
 :Windows,
 :WindowsRunMode,
 :Yaml]
```

Reply:
Sure. I will move it.